frontend/summary/summaryForms/prisma/PrismaArrowsTable.js

```diff
@@ -0,0 +1,146 @@
+import {inject, observer} from "mobx-react";
+import PropTypes from "prop-types";
+import React, {Component} from "react";
+import {ActionsTh, BaseEditableRow, MoveRowTd} from "shared/components/EditableRowData";
+import IntegerInput from "shared/components/IntegerInput";
+import SelectInput from "shared/components/SelectInput";
+import TextInput from "shared/components/TextInput";
+
+const key = "arrows";
+
+@inject("store")
+@observer
+class PrismaArrowsTable extends Component {
+    render() {
+        const items = this.props.store.subclass.settings[key],
+            {createNewArrow} = this.props.store.subclass;
+
+        return (
+            <div>
+                <h3>Arrows</h3>
+                <table className="table table-sm table-striped">
+                    {/* <colgroup>
+                    <col width="25%" />
+                    <col width="25%" />
+                    <col width="20%" />
+                    <col width="20%" />
+                    <col width="10%" />
+                </colgroup> */}
+                    <thead>
+                        <tr>
+                            <th>Source</th>
+                            <th>Destination</th>
+                            <th>Width</th>
+                            <th>Type</th>
+                            <th>Color</th>
+                            <ActionsTh onClickNew={createNewArrow} />
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {items.map((row, index) => {
+                            return <ArrowsRow row={row} index={index} key={index} />;
+                        })}
+                    </tbody>
+                </table>
+            </div>
+        );
+    }
+}
+
+@inject("store")
+@observer
+class ArrowsRow extends BaseEditableRow {
+    startEditable() {
+        return this.props.row.source == "";
+    }
+    renderViewRow(row, index) {
+        const {deleteArrayElement} = this.props.store.subclass;
+
+        return (
+            <tr>
+                <td>{row.source}</td>
+                <td>{row.dest}</td>
+                <td>{row.width}</td>
+                <td>{row.type}</td>
+                <td>{row.color}</td>
+                <MoveRowTd
+                    onDelete={() => deleteArrayElement(key, index)}
+                    onEdit={() => this.setState({edit: true})}
+                />
+            </tr>
+        );
+    }
+    renderEditRow(row, index) {
+        const {changeArraySettings, getArrowOptions} = this.props.store.subclass;
+        return (
+            <tr>
+                <td colSpan="100%">
+                    <div className="border my-2 p-2 pb-3 edit-form-background ">
```

Comment:
I would add a `pad-form` class to each edit row so that the inputs have a bit of separation